holoviews/core/data/grid.py

```diff
@@ -198,7 +198,7 @@ def coords(cls, dataset, dim, ordered=False, expanded=False, edges=False):
 
 
     @classmethod
-    def canonicalize(cls, dataset, data, coord_dims=None):
+    def canonicalize(cls, dataset, data, coord_dims=None, irregular_dims=[]):
```

Comment:
After discussing what ``irregular_dims`` really is, we agreed that we need a better name that makes it clearer that this is more of an xarray concept of dimensions than a holoviews one.

Reply:
Perhaps to make it clear,  this could be ``irregular_xrdim`` where ``xr`` helps show this is the xarray concept of dimensions. When dealing with xarray 'coords', that name should then be used to make the distinction clear.

Reply:
This is complicated by the fact this method doesn't only apply to xarray...so we might need some other suggestion.